src/debugger/direct/directDebugAdapter.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as path from "path";
+import { ReactNativeProjectHelper } from "../../common/reactNativeProjectHelper";
+import { ErrorHelper } from "../../common/error/errorHelper";
+import { getExtensionVersion } from "../../common/extensionHelper";
+import { ILaunchArgs } from "../../extension/launchArgs";
+import { getProjectRoot } from "../nodeDebugWrapper";
+import { Telemetry } from "../../common/telemetry";
+import { OutputEvent, Logger } from "vscode-debugadapter";
+import { TelemetryHelper } from "../../common/telemetryHelper";
+import { RemoteTelemetryReporter } from "../../common/telemetryReporters";
+import { ChromeDebugAdapter, ChromeDebugSession, IChromeDebugSessionOpts, IAttachRequestArgs, logger } from "vscode-chrome-debug-core";
+import { InternalErrorCode } from "../../common/error/internalErrorCode";
+import { RemoteExtension } from "../../common/remoteExtension";
+import { DebugProtocol } from "vscode-debugprotocol";
+import { getLoggingDirectory } from "../../extension/log/LogHelper";
+import * as nls from "vscode-nls";
+import * as Q from "q";
+const localize = nls.loadMessageBundle();
+
+export interface IDirectAttachRequestArgs extends IAttachRequestArgs, ILaunchArgs {
+    cwd: string; /* Automatically set by VS Code to the currently opened folder */
+}
+
+export interface IDirectLaunchRequestArgs extends DebugProtocol.LaunchRequestArguments, IDirectAttachRequestArgs { }
+
+export class DirectDebugAdapter extends ChromeDebugAdapter {
+
+    private outputLogger: (message: string, error?: boolean | string) => void;
+    private projectRootPath: string;
+    private remoteExtension: RemoteExtension;
+    private isSettingsInitialized: boolean; // used to prevent parameters reinitialization when attach is called from launch
```

Comment:
```suggestion
    private isSettingsInitialized: boolean; // used to prevent parameters reinitialization when attach is called from launch function
```